src/core/compile.c

```diff
@@ -657,7 +657,7 @@ static int macroexpand1(
     }
     Janet macroval;
     JanetBindingType btype = janet_resolve(c->env, name, &macroval);
-    if (btype != JANET_BINDING_MACRO ||
+    if (!(btype == JANET_BINDING_MACRO || btype == JANET_BINDING_DYNAMIC_MACRO) ||
```

Comment:
@pyrmont Could be

```c
(btype != JANET_BINDING_MACRO && btype != JANET_BINDING_DYNAMIC_MACRO)

as per DeMorgan's Law 🙂
```

Reply:
Yes, though I happen to find the current code more straight-forward to understand.  Perhaps that's not the case for everyone :)

Reply:
Really! Okay, suit yourself :slightly_smiling_face: